jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/UserInfo.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (C) 2011  JTalks.org Team
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package org.jtalks.jcommune.model.entity;
+
+import org.jtalks.common.model.entity.User;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Provides core user information which is later encapsulated
+ * into {@link Authentication} objects.
```

Comment:
`This class is immutable which is very important as it's going to be accessed from many threads. All the fields here are loaded only once into HTTP Session and don't ever change within that session.`

@oatkachenko have checked the functionality of changing the password?